plugins/lsp/conform-nvim.nix

```diff
@@ -0,0 +1,111 @@
+  ...
+}:
+with lib; let
+  cfg = config.plugins.conform-nvim;
+  helpers = import ../helpers.nix {inherit lib;};
+in {
+  options.plugins.conform-nvim =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "conform-nvim";
+
+      package = helpers.mkPackageOption "conform-nvim" pkgs.vimPlugins.conform-nvim;
+
+      # Map of filetype to formatters
+      formattersByFt =
+        helpers.defaultNullOpts.mkNullable (types.attrsOf types.anything)
+        "see documentation"
+        ''
+          # Map of filetype to formatters
+          formattersByFt =
+            {
+              lua = [ "stylua" ];
+              # Conform will run multiple formatters sequentially
+              python = [ "isort" "black" ];
+              # Use a sub-list to run only the first available formatter
+              javascript = [ [ "prettierd" "prettier" ] ];
+              # Use the "*" filetype to run formatters on all filetypes.
+              "*" = [ "codespell" ];
+              # Use the "_" filetype to run formatters on filetypes that don't
+              # have other formatters configured.
+              "_" = [ "trim_whitespace" ];
+             };
+        '';
+
+      formatOnSave =
+        helpers.defaultNullOpts.mkNullable (types.submodule {
+          options = {
+            lspFallback =
+              mkOption
+              {
+                type = types.bool;
+                description = "See :help conform.format for details.";
+              };
+            timeoutMs =
+              mkOption
+              {
+                type = types.bool;
+                description = "See :help conform.format for details.";
+              };
+          };
+        })
+        "see documentation"
+        ''
+          If this is set, Conform will run the formatter on save.
+          It will pass the table to conform.format().
+          This can also be a function that returns the table.
+          See :help conform.format for details.
+        '';
+
+      formatAfterSave =
+        helpers.defaultNullOpts.mkNullable (types.submodule {
+          options = {
+            lspFallback =
+              mkOption
+              {
+                type = types.bool;
+                description = "See :help conform.format for details.";
+              };
+          };
+        })
+        "see documentation"
+        ''
+          If this is set, Conform will run the formatter asynchronously after save.
+          It will pass the table to conform.format().
+          This can also be a function that returns the table.
+        '';
+
+      # Set the log level. Use `:ConformInfo` to see the location of the log file.
+      logLevel = helpers.defaultNullOpts.mkEnumFirstDefault ["ERROR" "DEBUG" "INFO" "TRACE" "WARN" "OFF"] "Set the log level. Use `:ConformInfo` to see the location of the log file.";
+
+      # Conform will notify you when a formatter errors
+      notifyOnError =
+        helpers.defaultNullOpts.mkBool true;
```

Comment:
this lacks a description, I meant to remove the Nix comment but leave the description string